lib/iris/_lazy_data.py

```diff
@@ -101,11 +101,14 @@ def array_masked_to_nans(array):
         result = array
     else:
         if ma.is_masked(array):
-            if array.dtype.kind == 'i':
-                array = array.astype(np.dtype('f8'))
             mask = array.mask
-            array[mask] = np.nan
-        result = array.data
+            if array.dtype.kind == 'i':
+                result = array.data.astype(np.dtype('f8'))
+            else:
+                result = array.data.copy()
+            result[mask] = np.nan
+        else:
+            result = array.data
```

Comment:
@lbdreyer This addresses the following numpy warning:
```
MaskedArrayFutureWarning: setting an item on a masked array which has a shared mask will not copy the mask and also change the original mask array in the future.
Check the NumPy 1.11 release notes for more information.
```
Which was generated by `array[mask] = np.nan` whenever the data was masked, but the `dtype` was not integral.